tsdb/engine/tsm1/engine.go

```diff
@@ -2204,6 +2202,33 @@ func (e *Engine) IteratorCost(measurement string, opt query.IteratorOptions) (qu
 	return cost, nil
 }
 
+func (e *Engine) seriesCost(seriesKey, field string, tmin, tmax int64) query.IteratorCost {
+	key := SeriesFieldKeyBytes(seriesKey, field)
+	c := e.FileStore.Cost(key, tmin, tmax)
+
+	// Retrieve the range of values within the cache.
+	cacheValues := e.Cache.Values(key)
```

Comment:
I think you can simplify this func to:

`c.CachedValues = len(e.Cache.Values(key).Include(tmin, tmax-1))`

That will filter return the count of values in the cache that are between [tmin, max).

Reply:
The maximum is inclusive so I'll just put `tmax`, but thanks! Done.